app/src/main/java/com/github/dedis/student20_pop/MainActivity.java

```diff
@@ -66,20 +71,41 @@ public void onClick(View view) {
                 break;
             case R.id.button_launch:
                 String name = ((EditText) findViewById(R.id.entry_box_launch)).getText().toString();
-                Person organizer = ((PoPApplication) getApplication()).getPerson();
+                final PoPApplication app = ((PoPApplication) getApplication());
                 // Creating the LAO and adding it to the organizer's LAO
-                Lao lao = new Lao(name, new Date(), organizer.getId());
-                organizer = organizer.setLaos(Collections.singletonList(lao.getId()));
+                Lao lao = new Lao(name, new Date(), app.getPerson().getId());
                 // Store the private key of the organizer
-                if (PrivateInfoStorage.storeData(this, organizer.getId(), organizer.getAuthentication()))
+                //TODO Move it into app onCreate()
+                if (PrivateInfoStorage.storeData(this, app.getPerson().getId(), app.getPerson().getAuthentication()))
                     Log.d(TAG, "Stored private key of organizer");
-                // TODO: send LAO and organizer information to backend
-                // Set LAO and organizer information locally
-                ((PoPApplication) getApplication()).setPerson(organizer);
-                ((PoPApplication) getApplication()).setLaos(Collections.singletonList(lao));
-                // Start the Organizer Activity (user is considered an organizer)
-                Intent intent = new Intent(this, OrganizerActivity.class);
-                startActivity(intent);
+
+                CompletableFuture.supplyAsync(() -> {
+                    try {
+                        //TODO Get URL dynamically
+                        return PoPClientEndpoint.connectToServer(URI.create("ws://10.0.2.2:2020/"), app.getPerson());
+                    } catch (DeploymentException e) {
+                        e.printStackTrace();
+                        return null;
+                    }
+                }).thenCompose(p -> {
+                    if(p != null)
+                        return p.createLoa(lao.getName(), lao.getTime(), lao.getTime(), app.getPerson().getId());
+                    else
+                        return null;
+                }).whenComplete((errCode, t) -> {
```

Comment:
What is errCode?

Reply:
The integer returned with the protocol. I am not sure it is needed in most cases as the request either failed and thus, t is not null. Or the request was a success and errCode is zero.